examples/asset-advanced/README.md

```diff
@@ -0,0 +1,240 @@
+This example shows usage of the asset module type with asset generator options customization.
+
+Files can be imported like other modules without file-loader or url-loader.
+
+# example.js
+
+```javascript
+import svg from "./images/file.svg";
+
+const container = document.createElement("div");
+Object.assign(container.style, {
+	display: "flex",
+	justifyContent: "center"
+});
+document.body.appendChild(container);
+
+function createImageElement(title, src) {
+	const div = document.createElement("div");
+	div.style.textAlign = "center";
+
+	const h2 = document.createElement("h2");
+	h2.textContent = title;
+	div.appendChild(h2);
+
+	const img = document.createElement("img");
+	img.setAttribute("src", src);
+	img.setAttribute("width", "150");
+	div.appendChild(img);
+
+	container.appendChild(div);
+}
+
+[svg].forEach(src => {
+	createImageElement(src.split(".").pop(), src);
+});
+```
+
+# webpack.config.js
+
+```javascript
+module.exports = {
+	output: {
+		assetModuleFilename: "images/[hash][ext]"
+	},
+	module: {
+		rules: [
+			{
+				test: /\.(png|jpg|svg)$/,
+				type: "asset",
+				generator: {
+					dataUrl: content => {
+						if (typeof content !== 'string') {
+							content = content.toString();
+						}
+
+						// just like in svg-url-loader https://github.com/bhovhannes/svg-url-loader/blob/e5ef4445f2fbb6f85245d8a6c88e2ea9a4377357/src/loader.js#L38-L44
```

Comment:
Maybe use https://www.npmjs.com/package/mini-svg-data-uri

Reply:
`mini-svg-data-uri` has too much lines of code, encoding in `svg-url-loader` is much shorter

Reply:
I don't mean to copy the code. I mean to used the package in the example, so when somebody copies the example they get a ready to use solution.

Reply:
Got it. Fixed